jdbc/src/main/java/com/interface21/jdbc/core/JdbcTemplate.java

```diff
@@ -1,17 +1,73 @@
 package com.interface21.jdbc.core;
 
+import com.interface21.dao.DataAccessException;
+import com.interface21.dao.IncorrectResultSizeDataAccessException;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+import javax.sql.DataSource;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
-import javax.sql.DataSource;
-
 public class JdbcTemplate {
 
     private static final Logger log = LoggerFactory.getLogger(JdbcTemplate.class);
 
     private final DataSource dataSource;
 
-    public JdbcTemplate(final DataSource dataSource) {
+    public JdbcTemplate(DataSource dataSource) {
         this.dataSource = dataSource;
     }
+
+    public int update(String sql, Object... args) {
+        Integer result = execute(sql, PreparedStatement::executeUpdate, args);
+        return Optional.ofNullable(result).orElse(0);
+    }
+
+    public <T> List<T> query(String sql, RowMapper<T> rowMapper, Object... args) {
+        return execute(sql, statement -> {
+            ArrayList<T> results = new ArrayList<>();
```

Comment:
```suggestion
            List<T> results = new ArrayList<>();
```
`ArrayList` 구현체로 받는 것보다 List 인터페이스로 받으면 어떨까요?

Reply:
해당 부분 반영하였습니다! 😃